sql/core/src/main/scala/org/apache/spark/sql/execution/command/ddl.scala

```diff
@@ -763,7 +763,9 @@ object DDLUtils {
   val HIVE_PROVIDER = "hive"
 
   def isHiveTable(table: CatalogTable): Boolean = {
-    table.provider.isDefined && table.provider.get.toLowerCase == HIVE_PROVIDER
+    // When `CatalogTable` is directly fetched from the catalog,
+    // CatalogTable.provider = None means the table is a Hive serde table.
+    !table.provider.isDefined || table.provider.get.toLowerCase == HIVE_PROVIDER
```

Comment:
The provider could be empty if the table is a VIEW. Thus, please do not modify the utility function here.